### ISSUE TITLE
Failure to tag route table causes delete fail

### ISSUE THREAD

**Issue by zmerlynn** [kubernetes/kops#597]
All `kubernetes-e2e-kops-aws-updown` jobs starting at [build 451](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws-updown/451) have been failing in the same way:

```
2016/10/05 05:59:16 e2e.go:669: Running: kops delete
TYPE        NAME                                ID
dhcp-options    e2e-kops-aws-updown.test-aws.k8s.io             dopt-4c30ad28
instance    master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io   i-3b161423
instance    master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io   i-4e0c0e56
instance    master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io   i-8705079f
instance    master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io   i-b70301af
instance    master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io   i-cf1210d7
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-1c050704
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-1d050705
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-aa1113b2
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-ab1113b3
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-b90301a1
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-ba0301a2
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-c80f0dd0
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-c90f0dd1
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-fe1311e6
instance    nodes.e2e-kops-aws-updown.test-aws.k8s.io           i-ff1311e7
vpc     e2e-kops-aws-updown.test-aws.k8s.io             vpc-84eaf5e0

instance:i-aa1113b2 ok
instance:i-3b161423 ok
vpc:vpc-84eaf5e0    still has dependencies, will retry
instance:i-1c050704 ok
instance:i-ff1311e7 ok
instance:i-ab1113b3 ok
instance:i-b90301a1 ok
instance:i-4e0c0e56 ok
instance:i-cf1210d7 ok
instance:i-c80f0dd0 ok
instance:i-fe1311e6 ok
instance:i-8705079f ok
instance:i-c90f0dd1 ok
instance:i-1d050705 ok
instance:i-b70301af ok
instance:i-ba0301a2 ok
Not all resources deleted; waiting before reattempting deletion
    dhcp-options:dopt-4c30ad28
    vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0    still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
    dhcp-options:dopt-4c30ad28
    vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0    still has dependencies, will retry
[...]
```

Additionally, it looks like we had a cataclysmic networking failure in [450](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws-updown/450), so I wonder if something was screwed up with the creation of that resource in the first place.

**Comment by zmerlynn** [comment]
By run [483](https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws-updown/483/build-log.txt), this is what we have:
```
2016/10/05 09:18:57 e2e.go:671: Step 'kops get' finished in 424.167307ms
2016/10/05 09:18:57 e2e.go:669: Running: kops delete
TYPE		NAME					ID
dhcp-options	e2e-kops-aws-updown.test-aws.k8s.io	dopt-4c30ad28
vpc		e2e-kops-aws-updown.test-aws.k8s.io	vpc-84eaf5e0

vpc:vpc-84eaf5e0	still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
	dhcp-options:dopt-4c30ad28
	vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0	still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
	dhcp-options:dopt-4c30ad28
	vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0	still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
	vpc:vpc-84eaf5e0
	dhcp-options:dopt-4c30ad28
vpc:vpc-84eaf5e0	still has dependencies, will retry
```

So the instances are gone by then, though interestingly they don't always appear to be gone, e.g. at [453](https://storage.googleapis.com/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws-updown/453/build-log.txt), after restarting delete, the instances are still around, even though we said `ok` during the last couple attempts to delete them:
```
2016/10/05 05:59:16 e2e.go:669: Running: kops delete
TYPE		NAME								ID
dhcp-options	e2e-kops-aws-updown.test-aws.k8s.io				dopt-4c30ad28
instance	master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io	i-3b161423
instance	master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io	i-4e0c0e56
instance	master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io	i-8705079f
instance	master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io	i-b70301af
instance	master-us-west-2a.masters.e2e-kops-aws-updown.test-aws.k8s.io	i-cf1210d7
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-1c050704
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-1d050705
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-aa1113b2
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-ab1113b3
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-b90301a1
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-ba0301a2
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-c80f0dd0
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-c90f0dd1
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-fe1311e6
instance	nodes.e2e-kops-aws-updown.test-aws.k8s.io			i-ff1311e7
vpc		e2e-kops-aws-updown.test-aws.k8s.io				vpc-84eaf5e0

instance:i-aa1113b2	ok
instance:i-3b161423	ok
vpc:vpc-84eaf5e0	still has dependencies, will retry
instance:i-1c050704	ok
instance:i-ff1311e7	ok
instance:i-ab1113b3	ok
instance:i-b90301a1	ok
instance:i-4e0c0e56	ok
instance:i-cf1210d7	ok
instance:i-c80f0dd0	ok
instance:i-fe1311e6	ok
instance:i-8705079f	ok
instance:i-c90f0dd1	ok
instance:i-1d050705	ok
instance:i-b70301af	ok
instance:i-ba0301a2	ok
Not all resources deleted; waiting before reattempting deletion
	dhcp-options:dopt-4c30ad28
	vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0	still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
	dhcp-options:dopt-4c30ad28
	vpc:vpc-84eaf5e0
vpc:vpc-84eaf5e0	still has dependencies, will retry
```

**Comment by zmerlynn** [comment]
With a little live debugging with @justinsb, we noticed the build 450 error:
```
I1005 05:38:48.781167    1018 executor.go:68] Tasks: 22 done / 47 total; 10 can run
W1005 05:41:03.770588    1018 executor.go:84] error running task "internetGateway/e2e-kops-aws-updown.test-aws.k8s.io": error adding tags to resource "igw-f615a592": Got retryable error while creating tags on "igw-f615a592", but retried too many times without success: InvalidInternetGatewayID.NotFound: The internetGateway ID 'igw-f615a592' does not exist
	status code: 400, request id: 32f25477-2f1d-41a8-b41f-cfc3acc3b47b
W1005 05:41:03.770623    1018 executor.go:84] error running task "routeTable/e2e-kops-aws-updown.test-aws.k8s.io": error adding tags to resource "rtb-2502dc42": Got retryable error while creating tags on "rtb-2502dc42", but retried too many times without success: InvalidRouteTableID.NotFound: The routeTable ID 'rtb-2502dc42' does not exist
	status code: 400, request id: 83ed41ae-c7eb-45f6-ab99-ac27b4a2d81a
I1005 05:41:03.770675    1018 executor.go:68] Tasks: 30 done / 47 total; 13 can run
I1005 05:41:04.423141    1018 apply_cluster.go:361] Adding docker image: https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-proxy.tar
I1005 05:41:04.425056    1018 apply_cluster.go:526] Found hash "1b13076fa602b1b3c6ef3f5f2b82e39ad0fe7424" for "https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-proxy.tar"
I1005 05:41:04.489623    1018 apply_cluster.go:361] Adding docker image: https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-proxy.tar
I1005 05:41:04.490724    1018 apply_cluster.go:526] Found hash "1b13076fa602b1b3c6ef3f5f2b82e39ad0fe7424" for "https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-proxy.tar"
I1005 05:41:04.490789    1018 apply_cluster.go:361] Adding docker image: https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-apiserver.tar
I1005 05:41:04.492028    1018 apply_cluster.go:526] Found hash "28d22ed4e68ade66a6023d709aa54b3647f8a61b" for "https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-apiserver.tar"
I1005 05:41:04.492060    1018 apply_cluster.go:361] Adding docker image: https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-controller-manager.tar
I1005 05:41:04.493253    1018 apply_cluster.go:526] Found hash "6a4abddfc4f2cdea11d2d4bfd12bb89974bedc0b" for "https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-controller-manager.tar"
I1005 05:41:04.493302    1018 apply_cluster.go:361] Adding docker image: https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-scheduler.tar
I1005 05:41:04.494419    1018 apply_cluster.go:526] Found hash "81f461e6210e5321da0ca419385adc8be8e404fe" for "https://storage.googleapis.com/kubernetes-release-dev/ci/v1.5.0-alpha.0.1915+0ad50d20339aec/bin/linux/amd64/kube-scheduler.tar"
W1005 05:41:04.793162    1018 executor.go:84] error running task "routeTable/e2e-kops-aws-updown.test-aws.k8s.io": error listing RouteTables: InvalidRouteTableID.NotFound: The routeTable ID 'rtb-2502dc42' does not exist
	status code: 400, request id: 828309b0-aabb-403a-bf2d-0b6677627bc8
W1005 05:41:04.793220    1018 executor.go:84] error running task "internetGateway/e2e-kops-aws-updown.test-aws.k8s.io": error listing InternetGateways: InvalidInternetGatewayID.NotFound: The internetGateway ID 'igw-f615a592' does not exist
	status code: 400, request id: 0b46ec77-8f69-4d0e-8f52-07532ce17776
I1005 05:41:04.793279    1018 executor.go:68] Tasks: 41 done / 47 total; 4 can run
```

**Comment by zmerlynn** [comment]
And here's the untagged route table:
```
        {   
            "Associations": [
                {   
                    "RouteTableAssociationId": "rtbassoc-62a8ae05",
                    "Main": true,
                    "RouteTableId": "rtb-2602dc41"
                }
            ],
            "RouteTableId": "rtb-2602dc41",
            "VpcId": "vpc-84eaf5e0",
            "PropagatingVgws": [],
            "Tags": [],
            "Routes": [
                {   
                    "GatewayId": "local",
                    "DestinationCidrBlock": "172.20.0.0/16",
                    "State": "active",
                    "Origin": "CreateRouteTable"
                }
            ]
        },
```

**Comment by zmerlynn** [comment]
Sorry, the one we were failing to tag is here:
```
{
    "RouteTables": [
        {
            "Associations": [], 
            "RouteTableId": "rtb-2502dc42", 
            "VpcId": "vpc-84eaf5e0", 
            "PropagatingVgws": [], 
            "Tags": [], 
            "Routes": [
                {
                    "GatewayId": "local", 
                    "DestinationCidrBlock": "172.20.0.0/16", 
                    "State": "active", 
                    "Origin": "CreateRouteTable"
                }, 
                {
                    "GatewayId": "igw-f615a592", 
                    "DestinationCidrBlock": "0.0.0.0/0", 
                    "State": "blackhole", 
                    "Origin": "CreateRoute"
                }
            ]
        }
    ]
}
```

**Comment by dmcnaught** [comment]
I think the below error is because you have to remove the dhcp-options set from the VPC before deleting the dhcp-options set and the VPC.
I get it everytime I delete with kops-1.5.0-beta1:
```bash
vpc:vpc-7f37a619	still has dependencies, will retry
Not all resources deleted; waiting before reattempting deletion
	vpc:vpc-7f37a619
	dhcp-options:dopt-8f63f2e8

Not making progress deleting resources; giving up
```

**Comment by chrislovecnm** [comment]
Has this been addressed?

**Comment by dmcnaught** [comment]
cluster create and delete works for me with kops 1.5.1 and k8s 1.5.2 and 1.5.3

**Comment by chrislovecnm** [comment]
@zmerlynn close?

**Comment by rkuzsma** [comment]
I've consistently had this issue in kops 1.5.1. I just upgraded to kops 1.5.3 and still seeing it there, too. Any more details I can provide to help diagnose?

**Comment by chrislovecnm** [comment]
@rkuzsma can you recreate with 1.6 alpha?  `kops delete cluster -v=10` would be awesome in a gist.  Also the details on how you setup the cluster.  Did you use your own vpc?  Did you create the vpc with kops? etc...

**Comment by rkuzsma** [comment]
@chrislovecnm As requested, here is a gist that shows the problem. Before creating the cluster tonight, I did a `brew install --HEAD kops`:
https://gist.githubusercontent.com/rkuzsma/38a149739e3d27bd0b12149ddb00e6f5/raw/ad9cebd8ad79bb09d0c398766dba5e8a9f510a0b/kops-output.txt

I create the cluster using `kops update cluster k8s.redacted.com --yes`.
After creating the cluster, I create some DNS A records in AWS Route53 for some nodes. I also open ports 80 and 443 ingress for the nodes.
I let kops provision the VPC.

**Comment by chrislovecnm** [comment]
@rkuzsma I am going tag this as blocks-next. Sorry to be specific about this, but --HEAD is a dev build, not a release.  Can you use the latest alpha release, by downloading the binary.

Please provide:
- kops create flags
- create log
- update log
- delete log.  

> After creating the cluster, I create some DNS A records in AWS Route53 for some nodes. I also open ports 80 and 443 ingress for the nodes.

Did you delete these resources before you removed the cluster?

@justinsb @geojaz any ideas on this??

**Comment by geojaz** [comment]
If you can share the kops create flags and create, update and delete logs- I can probably squash this one, this week.

**Comment by chrislovecnm** [comment]
@geojaz that is awesome, do you agree that this should block a release?

**Comment by geojaz** [comment]
@chrislovecnm Theoretically, yes ;) But I would really love to get a beta in the hands of users asap.

**Comment by geojaz** [comment]
It's probably going to need to be ported back to 1.5 as well...

**Comment by rkuzsma** [comment]
@chrislovecnm I've tried deleting the DNS AWS Route53 records and not closing the port 80 and 443 ports one time in the past, to no avail. I will retry that again with the Alpha release. I'll prepare the requested logs, but it will take me a little more time to run the specific binary (I'm on Mac, I need to provision a Linux VM) and to re-create the cluster from scratch with the same `instancegroup` configuration. This issue is not a blocker for me personally, just a minor annoyance.

**Comment by rkuzsma** [comment]
I can no longer reproduce this issue. I manually deleted the problematic DHCP and VPC, and now I can create and delete the cluster repeatedly without problems.

**Comment by geojaz** [comment]
@zmerlynn How are we doing on this? Do we still have work to do on this point or can we close this issue?

**Comment by geojaz** [comment]
this was fixed in around 1.5ish. closing it out
